### ISSUE TITLE
Design for on-hub notifications in the admin bar

### ISSUE THREAD

**Issue by hedgefield** [WordPress/wp-feature-notifications#23]
Continuing this on from #1, this is the place to discuss and iterate on the design of the notifications _in the admin bar_.

Below are the first few iterations already done. I will link to their original comments to preserve the context of the original posters:

- [Original designs by @hedgefield from 2018](https://github.com/WordPress/wp-notify/issues/1#issuecomment-679158048)
- [Fresh new 2020 design by @folletto and @ibdz](https://github.com/WordPress/wp-notify/issues/1#issuecomment-690501831)
- [Iteration by @jasmussen](https://github.com/WordPress/wp-notify/issues/1#issuecomment-693265889)
- [Iteration by @hedgefield](https://github.com/WordPress/wp-notify/issues/1#issuecomment-725508292)

I believe the first response was that the designs so far present [the functional requirements](https://docs.google.com/document/d/1SoIaFqXkXiVzq5mizbZQafMfL36bD0WKj8iwYM823MI/) of notifications as well as this specific UI usecase (a menu in the admin bar to view them) quite well already, but more feedback on that is always welcome, as well as nitty-gritty design detail discussion 🙂

**Comment by folletto** [comment]
Reposting this visual summary here for quick reference:

![types-overview-i1](https://user-images.githubusercontent.com/4389/99914589-4835e280-2cf6-11eb-9f64-0ddfc7d777b7.png)

**Comment by folletto** [comment]
Ok, I tried to blend together @jasmussen and @hedgefield's proposals. 

![wpnotify-notification-type-i4](https://user-images.githubusercontent.com/4389/99914496-b928ca80-2cf5-11eb-8e78-f395b3f60e6c.png)
 

Notes:

1. I kept Joen's bell as it seems a more standard approach
2. Preserved Tim's icon structure and approach
3. I tried to keep some consistency with Gutenberg's style as suggested
4. I opted for white because it seems the [most readable option](https://www.tandfonline.com/doi/abs/10.1080/00140139.2013.790485)
5. I tried to go for reduction and simplification, removing titles seems for example to bring more clarity and more clear messages, and moving the time below provides more evidence for the important part: the message.
6. I also opted for the "full height" approach (covering the admin bar) because it allows for more flexibility as well as creating a full height area of the screen that's more focused.
8. Colors / Text is using the native WP Admin styling, nothing new has been introduced
9. Accessibility wise, contrast is checked, and the semantic order has been borrowed from Tim's work as I think it's particularly effective: one accessing this panel will read "2 unread notifications", then gets the ability to clear all, and then can tab through the list. Not show here some kind of "Load X more" action at the end of the list in order to make the "Configure notification settings" accessible.

Please also note that the on-page equivalent will have to be reviewed too to be more aligned with the design choices made here.

**Comment by folletto** [comment]
For reference, here are the various configurations for the on-hub notifications:

![wpnotify-message-variants-i4](https://user-images.githubusercontent.com/4389/99915786-a7e3bc00-2cfd-11eb-8836-4f99cf9b1bde.png)

~This made me wonder if we need a title at all here. However: it might be necessary for the [on-page notifications](https://github.com/WordPress/wp-notify/issues/1#issuecomment-690501831), so I'll keep this question pending until we've explored these a bit better.~ → as expected the on-page notifications need it, see #26.

**Comment by jasmussen** [comment]
Solid work, folks 💕

**Comment by raaaahman** [comment]
Nice variants.

Although I thought that the logo (WordPress watermark here) **is** the image. Having one more image really cramps the space for the text in my opinion.

**Comment by folletto** [comment]
Yes, that's true — it's still up to the plugin author there, so I'd expect they'd balance out the need for an image with the text showing. 

I reached this approach trying to balance out the on-page ones in #26. That said, we could definitely decide that on-hub and on-page will require very different images to work. That might be an ok choice.

**Comment by raaaahman** [comment]
Maybe we could try something like:

- If notification has image, and image is from WordPress media Library, show the thumbnail image (or add a setting for even smaller images) left
- If notification has an image that exists in the plugin's file, show the image as is, left
- If notification hasn't any image, show sender's logo left

"Left" here refers what I see on the variants mock-up.

**Comment by folletto** [comment]
Yeah, ok! Message received: explore in i5 how it would work without a "right" image. ;)

**Comment by folletto** [comment]
Here's a concept exploration, "Bubble", I based on i4. I'm very aware it's not within the design guidelines of WordPress so it's not meant to land, but I still find valuable in throwing out ideas in case will then generate more ideas and out-of-the-box thinking (also note the white-on-coral-blue is not accessible).

![exploration-bubble-overlay](https://user-images.githubusercontent.com/4389/100524982-e32a3300-31b4-11eb-945c-73c727fa559f.png)

The behaviour would be similar as the slide-in from the right that happens for iPad multitasking.

**Comment by folletto** [comment]
Here's the tweaked i5 from the feedback above, not much style difference (just a tweak in the shadow for the scroll) but different handling of the image to align better with the on-page variant: 

![wpnotify-onhub-i5](https://user-images.githubusercontent.com/4389/100525397-6e58f800-31b8-11eb-881c-5a9a4d84129e.png)

![wpnotify-onhub-types-i5](https://user-images.githubusercontent.com/4389/100525399-70bb5200-31b8-11eb-8b6e-f197c5bb8334.png)

Check the mirror on-page design i5 on https://github.com/WordPress/wp-notify/issues/26#issuecomment-735287247.

**Comment by hedgefield** [comment]
Really good stuff, looks clean and I think ready for first implementation. Having the plugin icon as fallback for an optional image specified but showing them in the same location feels like a smart choice too. Cool!

**Comment by EdithAllison** [comment]
This appears superseded by the new design approach tracked in https://github.com/WordPress/wp-feature-notifications/issues/357, suggesting to close. @Sephsekla
